src/Console/Command/ListFilesCommand.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\ConfigInterface;
+use PhpCsFixer\Console\ConfigurationResolver;
+use PhpCsFixer\ToolInfoInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * @author Markus Staab <markus.staab@redaxo.org>
+ *
+ * @internal
+ */
+final class ListFilesCommand extends Command
+{
+    protected static $defaultName = 'list-files';
+
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    /**
+     * @var ToolInfoInterface
+     */
+    private $toolInfo;
+
+    public function __construct(ToolInfoInterface $toolInfo)
+    {
+        parent::__construct();
+
+        $this->defaultConfig = new Config();
+        $this->toolInfo = $toolInfo;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(
+                [
+                    new InputOption('config', '', InputOption::VALUE_REQUIRED, 'The path to a .php_cs file.'),
+                ]
+            )
+            ->setDescription('List all files beeing fixed by the given config.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $passedConfig = $input->getOption('config');
+
+        $resolver = new ConfigurationResolver(
+            $this->defaultConfig,
+            [
+                'config' => $passedConfig,
+            ],
+            getcwd(),
+            $this->toolInfo
+        );
+
+        $finder = $resolver->getFinder();
+
+        /** @var SplFileInfo $file */
+        foreach ($finder as $file) {
+            if ($file->isFile()) {
+                $output->writeln(escapeshellarg($file->getRelativePathname()));
```

Comment:
while real-world testing, I got 'sometimes' a error

```
...
'tests/Tokenizer/TransformersTest.php'
'tests/ToolInfoTest.php'
'tests/UtilsTest.php'
'tests/WhitespacesFixerConfigTest.php'
'tests/WordMatcherTest.php'
PHP Fatal error:  Uncaught Error: Call to undefined method SplFileInfo::getRelativePathname() in /mnt/c/dvl/GitHub/PHP-CS-Fixer/src/Console/Command/ListFilesCommand.php:86
Stack trace:
#0 /mnt/c/dvl/GitHub/PHP-CS-Fixer/vendor/symfony/console/Command/Command.php(255): PhpCsFixer\Console\Command\ListFilesCommand->execute()
#1 /mnt/c/dvl/GitHub/PHP-CS-Fixer/vendor/symfony/console/Application.php(971): Symfony\Component\Console\Command\Command->run()
#2 /mnt/c/dvl/GitHub/PHP-CS-Fixer/vendor/symfony/console/Application.php(290): Symfony\Component\Console\Application->doRunCommand()
#3 /mnt/c/dvl/GitHub/PHP-CS-Fixer/src/Console/Application.php(93): Symfony\Component\Console\Application->doRun()
#4 /mnt/c/dvl/GitHub/PHP-CS-Fixer/vendor/symfony/console/Application.php(166): PhpCsFixer\Console\Application->doRun()
#5 /mnt/c/dvl/GitHub/PHP-CS-Fixer/php-cs-fixer(102): Symfony\Component\Console\Application->run()
#6 {main}
  thrown in /mnt/c/dvl/GitHub/PHP-CS-Fixer/src/Console/Command/ListFilesCommand.php on line 86
```

its strange, since it works for nearly all files, but only a few error.
does the sfFinder sometimes return a `\SplFileInfo` and sometimes a `Symfony\Component\Finder\SplFileInfo`?
(`getRelativePathname` is only defined on the symfony-SplFileInfo class)

Reply:
that's probably not the only place, but looks like Finder can return built-in `\SplFileInfo` too
https://github.com/symfony/finder/blob/5.x/Finder.php#L660

Reply:
can you reproduce the issue with unit test?

Reply:
worst case scenario, simply check if object we got implements sf-file or \file and use one path or another